### DISCUSSION TITLE
Missing in reference table: com.google.common.util.concurrent.@Nullable

### DISCUSSION THREAD

**Discussion by tlopesPT** [smallrye/smallrye-open-api#1247] [Q&A] [answered]
Hey,

After upgrading the smallrye-open-api-maven-plugin to 3.0.0, I'm getting this error during mvn clean install:

```
Failed to execute goal io.smallrye:smallrye-open-api-maven-plugin:3.0.0:generate-schema (default) on project project-x: Could not generate OpenAPI Schema: java.lang.IllegalStateException: Missing in reference table: com.google.common.util.concurrent.@Nullable CollectionFuture$Present<V> ->
```

There's nothing special about the project and the error remains even after removing all openapi annotations.

I'm running Java 17 (Temurin on Mac M1 aarch64) and the project uses Quarkus 2.12.0.

I've got similar projects where the update to open-api 3.0.0 worked fine, just this one with the error.

Any help would be appreciated since the error is not pointing me in any particular direction.

**Comment by phillip-kruger** [comment]
Can you provide a small reproducer ?

**Reply by tlopesPT** [reply]
I've found that the error is caused by an Azure dependency:

    <dependency>
      <groupId>com.microsoft.azure.sdk.iot</groupId>
      <artifactId>iot-service-client</artifactId>
      <version>2.1.1</version>
    </dependency>

Adding this dependency along with the openapi mvn plugin to a brand new quarkus project with resteasy code sample should reproduce the error described.

I've uploaded a reproducer anyway: https://github.com/tlopesPT/smallrye-error

Still looking for a solution since I haven't found an obvious mismatch in that dependency, will start looking at exclusions via trial and error.

Edit:
The problem is being caused by Guava which is a dependency of the Azure sdk dependency above.

    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>31.1-jre</version>
      <scope>compile</scope>
    </dependency>

I've added Guava to the exclusions which fixed the problem, please tell me if there's a better approach or fix.

**Reply by MikeEdgar** [reply, marked as answer]
It looks like an issue in Jandex, I've opened https://github.com/smallrye/jandex/issues/256 to look into it.

**Reply by codespearhead** [reply]
@tlopesPT PR #256 is closed and included in v3.0.1 [as indicated in this comment](https://github.com/smallrye/jandex/issues/256#issuecomment-1249451674). Can you check if the problem persists in v3.11.0?

If it doesn't, can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so that it doesn't show up [in the discussion tab](https://github.com/smallrye/smallrye-open-api/discussions) as unanswered?
